### ISSUE TITLE
storage: TestReplicaLazyLoad timed out under stress

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#25430]
https://teamcity.cockroachdb.com/viewLog.html?buildId=650696&buildTypeId=Cockroach_UnitTests_Testrace&tab=artifacts

```
=== RUN   TestReplicaLazyLoad
W180510 14:15:23.049962 80556 storage/store.go:1516  [s1,r1/1:/M{in-ax}] could not gossip system config: periodic gossip is disabled
W180510 14:15:23.050105 80557 storage/store.go:1516  [s1,r1/1:/M{in-ax}] could not gossip node liveness: periodic gossip is disabled
I180510 14:15:23.050203 80336 storage/store.go:1424  [s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180510 14:15:23.050954 80336 gossip/gossip.go:333  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:43641" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W180510 14:15:23.059799 80476 storage/node_liveness.go:461  [hb] failed node liveness heartbeat: context deadline exceeded
W180510 14:15:23.065688 80481 storage/replica.go:3122  [s1,r1/1:/M{in-ax}] context canceled before proposing: 1 RequestLease
I180510 14:15:23.065898 80567 storage/client_test.go:1214  [hb,txn=68cc46e8] test clock advanced to: 0.018000125,0
I180510 14:15:23.066585 80476 storage/node_liveness.go:749  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (context finished during distsender send: context deadline exceeded)
W180510 14:15:23.068327 80476 storage/node_liveness.go:461  [hb] failed node liveness heartbeat: context deadline exceeded
I180510 14:15:23.074707 80476 storage/node_liveness.go:749  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (context finished during distsender send: context deadline exceeded)
W180510 14:15:23.076456 80476 storage/node_liveness.go:461  [hb] failed node liveness heartbeat: context deadline exceeded
W180510 14:15:23.082344 80476 storage/node_liveness.go:461  [hb] failed node liveness heartbeat: context deadline exceeded
I180510 14:15:23.088652 80476 storage/node_liveness.go:749  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (context finished during distsender send: context deadline exceeded)
W180510 14:15:23.090291 80476 storage/node_liveness.go:461  [hb] failed node liveness heartbeat: context deadline exceeded
I180510 14:15:23.097354 80476 storage/node_liveness.go:749  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (context finished during distsender send: context deadline exceeded)
I180510 14:15:23.098088 80476 storage/node_liveness.go:458  [hb] heartbeat failed on epoch increment; retrying
I180510 14:15:23.104698 80476 storage/node_liveness.go:749  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (context finished during distsender send: context deadline exceeded)
I180510 14:15:23.105673 80476 storage/node_liveness.go:458  [hb] heartbeat failed on epoch increment; retrying
[forever repeating]
```

**Comment by nvanbenschoten** [comment]
This is proving tricky to reproduce. I've run over 100,000 iterations without a failure.

@windchan7 https://github.com/cockroachdb/cockroach/pull/25281#issuecomment-389288523 indicates that you are easily able to reproduce this failure on that branch. Is that true?

**Comment by windchan7** [comment]
Yes, exactly. The main change I made that caused this failure was to use ondisk RocksDB engine for `multiTestContext `. Originally it's using inMem RocksDB.

**Comment by nvanbenschoten** [comment]
I talked this over with @windchan7 in person. We think the issue is that [node liveness update loop](https://github.com/cockroachdb/cockroach/blob/6d0c09aee9ff49ff0e6ba09bac86f5da90f9b63c/pkg/storage/node_liveness.go#L750) retries whenever it finds an `errRetryLiveness` error. Each of these liveness attempts are performed with the same `context`. We create these `errRetryLiveness` errors when we see [`AmbiguousResultErrors`](https://github.com/cockroachdb/cockroach/blob/6d0c09aee9ff49ff0e6ba09bac86f5da90f9b63c/pkg/storage/node_liveness.go#L833). Our theory is that a context cancellation is causing an `AmbiguousResultErrors` in DistSender, and this is triggering an endless loop that continues to use the cancelled context and continues to hit these `AmbiguousResultErrors`. The fix for this is trivial, and @windchan7 is testing with it in #25281.

**Comment by nvanbenschoten** [comment]
@windchan7 and I saw that this was still flaky. We worked through it and noticed that while #25631 had improved the situation and fixed a real bug, node liveness updates in the test were still continuously timing out. We realized that the test had [set its RaftTickInterval](https://github.com/cockroachdb/cockroach/blob/b1dd7ef11b19f295df28941658f5aa6062896cc8/pkg/storage/client_raft_test.go#L3097) so low that node liveness only had a few milliseconds to perform updates. We're testing now if bumping the interval to increase the amount of time node liveness has to perform updates will help.
